jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/IteratingCallback.java

```diff
@@ -179,8 +174,24 @@ protected void onSuccess()
     {
     }
 
+    /**
+     * Invoked when the overall task has been {@link #abort(Throwable) aborted} or {@link #failed(Throwable) failed}.
+     * <p>
+     * Calls to this method are serialized with respect to {@link #onAborted(Throwable)}, {@link #process()},
+     * {@link #onCompleteFailure(Throwable)} and {@link #onCompleted(Throwable)}.
+     *
+     * @param cause The cause of the failure or abort
+     */
+    protected void onFailure(Throwable cause)
```

Comment:
This method is called before completion, so it is wrong that it has replaced `onCompleteFailure()` almost everywhere, as most `onCompleteFailure()` implementations perform some buffer release (or similar) that are explicitly forbidden in this method.

Also, I'm not sure when could be safe to perform something from this method, because e.g. failing a callback from here may possibly result in re-entering, or calling some shared code that may assume that the buffer is released, or similar -- we call code that we don't know what it does.

Reply:
@sbordet The intention was to not change behaviour in this branch, as we tried that in #11876 and there were too many changes and we ended up playing whack-a-mole with bugs.    
But I'll create another branch without the renaming, to see what breaks.... my preference is still to progress with smaller PRs that fix one thing at a time, rather than changing every ICB usage (semantically) in one huge PR

Reply:
I have created #12055 to test this